src/test/datascience/notebook/helper.ts

```diff
@@ -255,9 +260,11 @@ async function waitForKernelToChangeImpl(
 
     // Find the kernel id that matches the name we want
     let id: string | undefined;
-    const labelOrId = 'labelOrId' in criteria ? criteria.labelOrId : undefined;
+    let labelOrId = 'labelOrId' in criteria ? criteria.labelOrId : undefined;
     if (labelOrId) {
-        id = notebookControllers?.find((k) => (labelOrId && k.label === labelOrId) || (k.id && k.id == labelOrId))?.id;
+        id = notebookControllers
+            ?.filter((k) => (criteria.isInteractiveController ? k.id.includes(InteractiveControllerIdSuffix) : true))
```

Comment:
I'm surprised our tests work today, 
Either they are not working correctly or we don't have tests that change kernels in IW.
Because the old code didn't work at all, we'd end up getting a controller that was dedicated for IPynb Notebooks.
This ensures we use the IW specific controller.